nasa/harmony
# Harmony 997 prestop hooks

Add PreStop hooks to prevent pod termination while work is being done. 
To test 
1. checkout this branch and do a 
2.build the services image (`npm run build` in tasks/service-runner)
3. start everything locally using `~/bin/start-argo` and ~`/bin/deploy-services`
4. run `kubectl get pods -n argo` to get the list of pods running
5. run `kubectl logs -f <harmony service example pod> manager` to tail the logs of the pod
6. use the following query to start a relatively long running granule 
```
http://internal-harmony-jn-997-frontend-1497576956.us-west-2.elb.amazonaws.com/C1233800302-EEDTEST/ogc-api-coverages/1.0.0/collections/all/coverage/rangeset?granuleId=G1233800343-EEDTEST,G1233800344-EEDTEST&format=image%2Fpng&maxResults=1&turbo=true
```
7. once you see the logs indicate work is being performed, try to delete the pod
`kubectl delete pod <harmony service example pod>`

you should see the query complete before the pod gets deleted
